src/dodal/common/crystal_metadata.py

```diff
@@ -0,0 +1,61 @@
+import math
+from dataclasses import dataclass
+from enum import Enum
+from typing import Literal
+
+
+@dataclass(frozen=True)
+class Material:
+    """
+    Class representing a crystalline material with a specific lattice parameter.
+    """
+
+    name: str
+    lattice_parameter: float  # Lattice parameter in meters
+
+
+class MaterialsEnum(Enum):
+    Si = Material(name="silicon", lattice_parameter=5.4310205e-10)
+    Ge = Material(name="germanium", lattice_parameter=5.6575e-10)
+
+
+@dataclass(frozen=True)
+class CrystalMetadata:
+    """
+    Metadata used in the NeXus format,
+    see https://manual.nexusformat.org/classes/base_classes/NXcrystal.html
+    """
+
+    usage: Literal["Bragg", "Laue"]
+    type: str
+    reflection: tuple[int, int, int]
+    d_spacing: tuple[float, str]
+
+    @staticmethod
+    def calculate_default_d_spacing(
```

Comment:
Should: Sorry, maybe I've misunderstood, I thought the conclusion of the discussion at https://github.com/DiamondLightSource/dodal/pull/843#discussion_r1827660919 was to remove all this calculation and use the PV? Or is it not yet on i22?

Reply:
If the PV isn't on i22 yet then I said in that comment, can we get the ball rolling on that?

Reply:
there was a discussion was to add the PV on i18, which ultimately just was a slight name change.

regarding the static method
>             self.crystal_metadata_d_spacing = epics_signal_r(float, "DSPACING:RBV")

at the moment there is the PV value used. I might have mistaken with the line above where the 'initial value' is used.

at the moment the only place when the calculation is used is in the `make_crystal_metadata_from_material` where if the d_spacing is not supplied by the user, it's calculated. 

we could remove it and make it required

Reply:
Ok, so we still need it in there for i22. That's fine, I'll make a note on the conversion issue.